pysoa/test/stub_service.py

```diff
@@ -353,14 +353,15 @@ def call_body(self):
         def call_bodies(self):
             return tuple(args[0][0] for args in self.call_args_list)
 
-    def __init__(self, service, action, body=None, errors=None):
+    def __init__(self, service, action, body=None, errors=None, side_effect=None):
         assert isinstance(service, six.text_type), 'Stubbed service name "{}" must be unicode'.format(service)
         assert isinstance(action, six.text_type), 'Stubbed action name "{}" must be unicode'.format(action)
 
```

Comment:
You need an `assert not (body or actions) or not side_effect`. It's not valid to provide both.